streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignorTest.java

```diff
@@ -435,7 +447,7 @@ public void testEagerSubscription() {
         // When following the eager protocol, we must encode the previous tasks ourselves since we must revoke
         // everything and thus the "ownedPartitions" field in the subscription will be empty
         final SubscriptionInfo info = getInfo(processId, prevTasks, standbyTasks, null);
-        assertEquals(info.encode(), subscription.userData());
+        assertEquals(info, SubscriptionInfo.decode(subscription.userData()));
```

Comment:
Changing these assertions to compare SubscriptionInfo objects instead of byte arrays, so that when the tests fail, we can actually see why.

If anyone really wants to compare the byte arrays, I'll write a utility method to compare the bytes, but print the message containing the objects.

Reply:
+1. All for user readable assertions. If the bytes format is standardized then it should be tested in SubscriptionInfoTest.